src/deepsparse/transformers/haystack/pipeline.py

```diff
@@ -0,0 +1,517 @@
+    :param input_shapes: list of shapes to set ONNX the inputs to. Pass None
+        to use model as-is. Default is None
+    :param alias: optional name to give this pipeline instance, useful when
+        inferencing with multiple models. Default is None
+    :param sequence_length: sequence length to compile model and tokenizer for.
+        If a list of lengths is provided, then for each length, a model and
+        tokenizer will be compiled capable of handling that sequence length
+        (also known as a bucket). Default is 128
+    :param docs: list of documents to be written to document_store. Can also
+        be written after instantiation with write_documents method.
+        Default is None
+    :param config: dictionary or instance of HaystackPipelineConfig. Used to
+        specify Haystack node arguments
+    :param retriever_kwargs: keyword arguments to be passed to retriever. If
+        the retriever is a deepsparse retriever, then these arguments will also
+        be passed to the EmbeddingExtractionPipeline of the retriever
+    """
+
+    def __init__(
+        self,
+        *,
+        engine_type: str = DEEPSPARSE_ENGINE,
+        batch_size: int = 1,
+        num_cores: Optional[int] = None,
+        scheduler: Optional[Scheduler] = None,
+        input_shapes: List[List[int]] = None,
+        alias: Optional[str] = None,
+        context: Optional[Context] = None,
+        sequence_length: Optional[int] = None,
+        docs: Optional[List[Dict]] = None,
+        config: Optional[Union[HaystackPipelineConfig, Dict[str, Any]]] = None,
+        **retriever_kwargs,
+    ):
+        # transformer pipeline members
+        self._sequence_length = sequence_length or 128
+        self.config = None
+        self.tokenizer = None
+        self.onnx_input_names = None
+        self._temp_model_directory = None
+
+        # pipeline members
+        self._model_path_orig = None
+        self._model_path = None
+        self._engine_type = engine_type
+        self._batch_size = batch_size
+        self._alias = alias
+
+        if retriever_kwargs.get("batch_size") and retriever_kwargs["batch_size"] != 1:
+            raise ValueError(
+                f"{self.__class__.__name__} currently only supports batch size 1, "
+                f"batch size set to {retriever_kwargs['batch_size']}"
+            )
+
+        # pass arguments to retriever (which then passes to extraction pipeline)
+        retriever_kwargs["engine_type"] = engine_type
+        retriever_kwargs["batch_size"] = batch_size
+        retriever_kwargs["num_cores"] = num_cores
+        retriever_kwargs["scheduler"] = scheduler
+        retriever_kwargs["input_shapes"] = input_shapes
+        retriever_kwargs["alias"] = alias
+        retriever_kwargs["context"] = context
+        retriever_kwargs["sequence_length"] = self._sequence_length
+        self._config = self._parse_config(config)
+
+        self.initialize_pipeline(retriever_kwargs)
```

Comment:
add placeholders here for any class members that will be populated by `initialize_pipeline` (ie `self._retriever = None`)